kerby-kerb/kerb-client/src/main/java/org/apache/kerby/kerberos/kerb/client/impl/AbstractInternalKrbClient.java

```diff
@@ -153,13 +153,14 @@ public SgtTicket requestSgt(KOptions requestOptions) throws KrbException {
                 getStringOption(KrbOption.SERVER_PRINCIPAL));
         PrincipalName serverPrincipalName = new PrincipalName(serverPrincipalString);
 
+        PrincipalName clientPrincipalName = null;
         if (tgtTicket != null) {
             String sourceRealm = tgtTicket.getRealm();
             String destRealm = serverPrincipalName.getRealm();
             if (!sourceRealm.equals(destRealm)) {
                 KrbConfig krbConfig = krbSetting.getKrbConfig();
                 LinkedList<String> capath = krbConfig.getCapath(sourceRealm, destRealm);
-                PrincipalName clientPrincipalName = tgtTicket.getClientPrincipal();
+                clientPrincipalName = tgtTicket.getClientPrincipal();
```

Comment:
This line doesn't apply if the source realm is equal to the dest realm, so you could move it above the if statement.

Reply:
right, I'll fix